Code/Framework/AzToolsFramework/AzToolsFramework/ViewportUi/ViewportUiDisplay.cpp

```diff
@@ -303,42 +305,78 @@ namespace AzToolsFramework::ViewportUi::Internal
         return false;
     }
 
+    QMargins ViewportUiDisplay::CalculateViewportElementMargins() const
+    {
+        if (m_imGuiActive)
+        {
+            if (GetViewportBorderVisible())
+            {
+                return ViewportUiOverlayImGuiBorderMargin;
+            }
+            else
+            {
+                return ViewportUiOverlayImGuiMargin;
+            }
+        }
+        else
+        {
+            if (GetViewportBorderVisible())
+            {
+                return ViewportUiOverlayBorderMargin;
+            }
+            else
+            {
+                return ViewportUiOverlayDefaultMargin;
+            }
+        }
+    }
+
     void ViewportUiDisplay::CreateViewportBorder(
         const AZStd::string& borderTitle, AZStd::optional<ViewportUiBackButtonCallback> backButtonCallback)
     {
         m_uiOverlay.setStyleSheet(QString("border: %1px solid %2; border-top: %3px solid %4;")
                                       .arg(
                                           QString::number(HighlightBorderSize), HighlightBorderColor,
                                           QString::number(ViewportUiTopBorderSize), HighlightBorderColor));
-        m_uiOverlayLayout.setContentsMargins(
-            HighlightBorderSize + ViewportUiOverlayMargin, ViewportUiTopBorderSize + ViewportUiOverlayMargin,
-            HighlightBorderSize + ViewportUiOverlayMargin, HighlightBorderSize + ViewportUiOverlayMargin);
+
         m_viewportBorderText.setAlignment(Qt::AlignCenter);
 
         m_viewportBorderText.show();
+        m_uiOverlayLayout.setContentsMargins(CalculateViewportElementMargins());
         ChangeViewportBorderText(borderTitle.c_str());
 
         // only display the back button if a callback was provided
         m_viewportBorderBackButtonCallback = backButtonCallback;
         m_viewportBorderBackButton.setVisible(m_viewportBorderBackButtonCallback.has_value());
     }
 
-    void ViewportUiDisplay::ChangeViewportBorderText(
-        const char* borderTitle)
+    void ViewportUiDisplay::ChangeViewportBorderText(const char* borderTitle)
     {
         // when the text changes if the width is different it will flicker as it changes,
         // this sets the width to the entire overlay to avoid that
         m_viewportBorderText.setFixedWidth(m_uiOverlay.width());
         m_viewportBorderText.setText(borderTitle);
     }
 
+    void ViewportUiDisplay::ImGuiActive(bool active)
+    {
+        if (active)
```

Comment:
Suggestion: `m_imGuiActive = active;`

Reply:
Might also want to add `if (active != m_imGuiActive)` first too (not essential though)